cpu/stm32_common/periph/rtc.c

```diff
@@ -322,4 +326,231 @@ void ISR_NAME(void)
     cortexm_isr_end();
 }
 
+#elif defined(CPU_FAM_STM32F1)
```

Comment:
I haven't looked in detail if some parts could actually be shared with the existing implementation but if it's not the case, it would be better to have a F1 specific rtc.c file in cpu/stm32f1/periph/rtc.c. If it's possible to make the initial common driver more generic, then I would try to adapt it.